### ISSUE TITLE
[🐛 Bug]: C# - DevToolsSession - Close() last tab causes unhandeled exception crash

### ISSUE THREAD

**Issue by schrufygroovy** [SeleniumHQ/selenium#9973]
### What happened?

When one has `.GetDevToolsSession()` used once, and then one uses `.Close()` at the last tab of the browser, an unhandeled exception occures, that is impossible to catch for the owner of the program, because it happens in `OnTargetDetached` event handler of `OpenQA.Selenium.DevTools.DevToolsSession`.

On local chrome driver I receive this error:
```
System.InvalidOperationException: A command response was not received: Target.getTargets
   at OpenQA.Selenium.DevTools.DevToolsSession.SendCommand(String commandName, JToken params, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
   at OpenQA.Selenium.DevTools.DevToolsSession.SendCommand[TCommand,TCommandResponse](TCommand command, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
   at OpenQA.Selenium.DevTools.V95.Target.TargetAdapter.GetTargets(GetTargetsCommandSettings command, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
   at OpenQA.Selenium.DevTools.V95.V95Target.GetTargets()
   at OpenQA.Selenium.DevTools.DevToolsSession.InitializeSession()
   at OpenQA.Selenium.DevTools.DevToolsSession.OnTargetDetached(Object sender, TargetDetachedEventArgs e)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__140_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

### Additional Info

The bug was probably introduce by the changes mentioned in this comment: https://github.com/SeleniumHQ/selenium/issues/9798#issuecomment-929411979 .

On remote driver I receive a slightly different error in the same `OnTargetDetached` function:
```
System.InvalidOperationException: A command response was not received: Target.setAutoAttach
     at OpenQA.Selenium.DevTools.DevToolsSession.SendCommand(String commandName, JToken params, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
     at OpenQA.Selenium.DevTools.DevToolsSession.SendCommand[TCommand,TCommandResponse](TCommand command, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
     at OpenQA.Selenium.DevTools.V95.Target.TargetAdapter.SetAutoAttach(SetAutoAttachCommandSettings command, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
     at OpenQA.Selenium.DevTools.V95.V95Target.SetAutoAttach()
     at OpenQA.Selenium.DevTools.DevToolsSession.InitializeSession()
     at OpenQA.Selenium.DevTools.DevToolsSession.OnTargetDetached(Object sender, TargetDetachedEventArgs e)
```

### How can we reproduce the issue?

Program.cs

```
using System;
using System.Threading.Tasks;
using OpenQA.Selenium.Chrome;

using (var driver = new ChromeDriver())
{
    driver.Url = "https://www.selenium.dev";
    var devToolsSession = driver.GetDevToolsSession();
    driver.Close();
    await Task.Delay(TimeSpan.FromSeconds(5));
}

```

Project file

```
<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>net5.0</TargetFramework>
    </PropertyGroup>

    <ItemGroup>
      <PackageReference Include="Selenium.WebDriver" Version="4.0.0" />
      <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
      <PackageReference Include="Selenium.WebDriver.ChromeDriver" Version="95.0.4638.1700" />
    </ItemGroup>

</Project>
```

https://github.com/schrufygroovy/devtoolssessionclosecrashes

### Relevant log output

```shell
System.InvalidOperationException: A command response was not received: Target.getTargets
   at OpenQA.Selenium.DevTools.DevToolsSession.SendCommand(String commandName, JToken params, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
   at OpenQA.Selenium.DevTools.DevToolsSession.SendCommand[TCommand,TCommandResponse](TCommand command, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
   at OpenQA.Selenium.DevTools.V95.Target.TargetAdapter.GetTargets(GetTargetsCommandSettings command, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
   at OpenQA.Selenium.DevTools.V95.V95Target.GetTargets()
   at OpenQA.Selenium.DevTools.DevToolsSession.InitializeSession()
   at OpenQA.Selenium.DevTools.DevToolsSession.OnTargetDetached(Object sender, TargetDetachedEventArgs e)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__140_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

### Operating System

Windows 10

### Selenium version

4.0.0

### What are the browser(s) and version(s) where you see this issue?

Chrome 95.0.4638.54

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver 95.0.4638.1700

### Are you using Selenium Grid?

Similar error on RemoteDriver but slightly different
```
System.InvalidOperationException: A command response was not received: Target.setAutoAttach
     at OpenQA.Selenium.DevTools.DevToolsSession.SendCommand(String commandName, JToken params, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
     at OpenQA.Selenium.DevTools.DevToolsSession.SendCommand[TCommand,TCommandResponse](TCommand command, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
     at OpenQA.Selenium.DevTools.V95.Target.TargetAdapter.SetAutoAttach(SetAutoAttachCommandSettings command, CancellationToken cancellationToken, Nullable`1 millisecondsTimeout, Boolean throwExceptionIfResponseNotReceived)
     at OpenQA.Selenium.DevTools.V95.V95Target.SetAutoAttach()
     at OpenQA.Selenium.DevTools.DevToolsSession.InitializeSession()
     at OpenQA.Selenium.DevTools.DevToolsSession.OnTargetDetached(Object sender, TargetDetachedEventArgs e)
```

**Comment by jimevans** [comment]
@schrufygroovy What happens if you use `4.0.1`?

**Comment by schrufygroovy** [comment]
@jimevans - bug is fixed with `4.0.1`
